ddht/v5_1/alexandria/client.py

```diff
@@ -598,6 +603,10 @@ async def _feed_talk_requests(self) -> None:
                 except DecodingError:
                     pass
                 else:
+                    if message.type != AlexandriaMessageType.REQUEST:
+                        self.logger.warning("Received non-REQUEST msg via TALKREQ: %s", message)
```

Comment:
In the issue description you mentioned avoiding the decoding if the msg is of the wrong type, but I couldn't find a way to achieve that since the type is included in the underlying msg's payload.